src/pages/api/teams/create.ts

```diff
@@ -0,0 +1,101 @@
+import ShortUniqueId from "short-unique-id";
+import { createClient } from "@supabase/supabase-js";
+import { sendEmail } from "~/pages/api/utils/mailer.ts";
+import type { APIRoute } from "astro";
+import type { CreateTeamItem } from "~/types";
+
+export const prerender = false;
+
+const supabase = createClient(
+  import.meta.env.SUPABASE_URL,
+  import.meta.env.SUPABASE_ANON_KEY,
+);
+
+const { randomUUID } = new ShortUniqueId({ length: 6 });
+
+export const POST: APIRoute = async ({ request }) => {
+  const formData = await request.formData();
+
+  let errors: String[] = [];
+
+  // forms validation
+  const valid = await validateForms(formData, errors);
+  if (!valid) {
+    return new Response(
+      JSON.stringify({
+        message: { errors: errors },
+        status: 400,
+      }),
+      { status: 400 },
+    );
+  }
+
+  const insertion_data: CreateTeamItem = {
+    team_name: formData.get("name")?.toString() ?? "",
+    member_email: formData.get("email")?.toString() ?? "",
+    new_team_code: randomUUID().toUpperCase()
+  };
+
+  const { data, error } = await supabase.rpc("create_team", insertion_data);
+
+  if (error) {
+    console.error(error);
+    errors.push("There was an error with your credentials. Check if you are already in a team or if you are not registered.");
+
+    return new Response(
+      JSON.stringify({
+        message: { errors: errors },
+        status: 500,
+      }),
+      { status: 500 },
+    );
+  }
+
+  sendTeamCreationEmail(insertion_data.member_email, insertion_data.team_name, insertion_data.new_team_code);
+
+  return new Response(
+    JSON.stringify({
+      status: 200,
+    }),
+    { status: 200 },
+  );
+};
+
+const validateForms = async (formData: FormData, errors: String[]) => {
+  let { data: participants, error } = await supabase
+    .from("participants")
+    .select("email, confirmation")
+    .eq("email", formData.get("email")?.toString() ?? "")
+    .eq("confirmation", formData.get("confirmation")?.toString() ?? "");
+
+  let valid = true;
+  if (participants && participants.length === 0) {
+    valid = false;
+    errors.push("Participant not registered or confirmation code does not match");
```

Comment:
Don't this error state the same as the one on line 43?

Reply:
You're right. Since I'm making that check before calling the postgres function, that won't be the error. I'll make the error on line 43 more generic, since the cause of it could be a variety of things.